File: StackOverflow/StackOverflow.java
public class StackOverflow {
	public static int fonction( int n){
		if(n <= 0){
			return 0;
		}
		else{
			return 1+fonction(n-1);
		}

	}
	public static void main(String[] args){
			System.out.println(fonction(1000));
			System.out.println(fonction(1000000));
	
	} 

}

File: StackOverflow/stack_overflow.c
#include <stdio.h>

int fonction( int n){

	if(n<=0){
		return(0);	
	}
		
	if(n>=1){
	return (1+fonction(n-1));
	}
		
}

int main (void){
	int res1;
	int res2;

	res1 =fonction(1000);
	res2 = fonction(1000000);

	printf("res1 = %d",res1);
	printf("res2 = %d",res2);
}
